dashboard/lti_new.py

```diff
@@ -0,0 +1,197 @@
+    elif course_details.term_id is not None:
+        logger.info(f'Course {course_id} data is ready')
+        is_course_data_loaded = True
+
+    myla_globals = {
+        "username": username,
+        "is_superuser": user_obj.is_staff,
+        "login": settings.LOGIN_URL,
+        "logout": settings.LOGOUT_URL,
+        "user_image": user_img,
+        "primary_ui_color": settings.PRIMARY_UI_COLOR,
+        "help_url": settings.HELP_URL,
+        "google_analytics_id": settings.GA_ID,
+        "user_courses_info": [
+            {"course_id": course_id, "course_name": course_name, "enrollment_type": short_user_role_list(user_roles)}],
+        "lti_launch_id": launch_id,
+        "lti_is_course_data_loaded": is_course_data_loaded,
+    }
+    return myla_globals
+
+
+@csrf_exempt
+def login(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'LTI Login failed due to {e}'
+        }
+        return JsonResponse(error_message, status=500)
+
+    target_link_uri = request.POST.get('target_link_uri', request.GET.get('target_link_uri'))
+    if not target_link_uri:
+        error_message = {
+            'lti_error_message': 'LTI Login failed due to missing "target_link_uri" param'
+        }
+        return JsonResponse(error_message, status=500)
+    dummy_cache = DUMMY_CACHE in settings.DB_CACHE_CONFIGS['BACKEND']
+    launch_data_storage = DjangoCacheDataStorage(cache_name='default') if not dummy_cache else None
+    oidc_login = DjangoOIDCLogin(request, tool_conf, launch_data_storage=launch_data_storage)
+    return oidc_login.enable_check_cookies().redirect(target_link_uri)
+
+
+@require_POST
+@csrf_exempt
+def launch(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'LTI Launch failed due to {e}'
+        }
+        return JsonResponse(error_message, status=500)
+    dummy_cache = DUMMY_CACHE in settings.DB_CACHE_CONFIGS['BACKEND']
+    launch_data_storage = DjangoCacheDataStorage(cache_name='default') if not dummy_cache else None
+    message_launch = DjangoMessageLaunch(request, tool_conf, launch_data_storage=launch_data_storage)
+    if not dummy_cache:
+        cache_ttl = settings.DB_CACHE_CONFIGS['CACHE_TTL']
+        cache_lifetime = cache_ttl if cache_ttl else 7200
+        # fetch platform's public key from cache instead of calling the API will speed up the launch process
+        message_launch.set_public_key_caching(launch_data_storage, cache_lifetime=cache_lifetime)
+    try:
+       myla_globals = extracting_launch_variables_for_tool_use(request, message_launch)
+    except Exception as e:
+        error_message = {
+            'lti_error_message': f'LTI Launch failed due to {e}'
+        }
+        return JsonResponse(error_message, status=500)
```

Comment:
This section is included in the one repeated above, but it's used under different circumstances. (Not trying to use `get_tool_conf()` here.)   I think this error message and status code could be refactored into a new class based on `JsonResponse`.  It could be a subclass or composition.  The new class would let you simply call it as:

```python
return LtiError(lti_exception)
```

It should handle the message formatting and set the status code.